### ISSUE TITLE
[Question] Is `onIdTokenChanged` still necessary?

### ISSUE THREAD

**Issue by steve-marmalade** [awinogrodzki/next-firebase-auth-edge#178]
Hey @awinogrodzki , hope you've been well!

I was reading through your updated documentation to make sure I hadn't missed any new features and noticed this line in [Authentication Middleware](https://next-firebase-auth-edge-docs.vercel.app/docs/usage/middleware)
> Automatically refreshes browser authentication cookies when token expires, and let's developer act accordingly

If the middleware takes care of checking whether the token has expired (and automatically refreshes it), do I still need a Client component that registers an event listener for `onIdTokenChanged` and calls `/api/login`?

**Comment by awinogrodzki** [comment]
Hey @steve-marmalade! 

That's a great question.

Basically yes, you don't need to use `onIdTokenChanged`. The new starter example also does not use it anymore. It just passes a token from server. See: https://github.com/awinogrodzki/next-firebase-auth-edge/issues/143

I imagine there might be cases where `onIdTokenChanged` is helpful. Eg. client-side account delete detection. It's up to developer whether to use it or not.

TLDR; Don't use it if you're not use if you need it

Cheers 🍻

**Comment by awinogrodzki** [comment]
I will close the issue now. Let me know if the above answers your question
